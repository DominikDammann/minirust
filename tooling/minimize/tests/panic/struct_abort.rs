//@ compile-flags: -C panic=abort

extern crate intrinsics;
use intrinsics::*;

struct A {
    x: i32,
}

impl Drop for A {
    fn drop(&mut self) {
        print(self.x);
    }
}

#[allow(unconditional_recursion)]
fn f(elem: A) {
    print(100 / elem.x); // Causes a panic if `elem.x == 0`
    let next = A { x: elem.x - 1 };
    f(next);
    print(-1); // Unreachable
}

fn main() {
    let a = A { x: 5 };
    f(a);
    print(-1); // Unreachable
}
