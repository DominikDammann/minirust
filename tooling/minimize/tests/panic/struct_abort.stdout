20
25
33
50
100
